### ISSUE TITLE
Ideas for the project

### ISSUE THREAD

**Issue by n000tavailable** [n000tavailable/tunedetective#5]
**TL;DR: Share your ideas to improve the app (and maintain privacy.)**

Dear users,

I need your help! I'm reaching out to all of you who frequently use the app and have a deep interest in its development. Your valuable ideas and suggestions are crucial to enhance its features and overall experience.

I'm running low on ideas and would greatly appreciate your input. If you have any fresh ideas or innovative concepts that you'd like to see implemented, please let me know. I'm open to your suggestions and eager to explore new possibilities.

Please keep in mind that I am committed to maintaining a privacy-friendly focus. This means avoiding the need for user accounts or interactions with non-open source projects. Your privacy is of utmost importance to me.

Together, we can shape this app into something truly exceptional. Your contributions, whether in the form of ideas or code, can have a profound impact on its future.

Thank you for considering my request. I look forward to hearing from you and working together to make this app even better.

Warm regards,
Jonas

**Comment by lindely** [comment]
Hi,

I used to follow some artists on Twitter to keep up with their releases, but then Musk happened. So the idea behind the app really connects with a need I started to develop.

When I opened the app and entered some artist names, the first thing I thought was "Why can I only see the last release?". So, without having a clue on how difficult or impossible this may be, the one thing I would love to see is an option to scroll through previous releases of artists.

Thanks for developing the app :-)

**Comment by n000tavailable** [comment]
> Hi,
> 
> I used to follow some artists on Twitter to keep up with their releases, but then Musk happened. So the idea behind the app really connects with a need I started to develop.
> 
> When I opened the app and entered some artist names, the first thing I thought was "Why can I only see the last release?". So, without having a clue on how difficult or impossible this may be, the one thing I would love to see is an option to scroll through previous releases of artists.
> 
> Thanks for developing the app :-)

I wanted to update you on the progress of your idea to view an artist's discography. Good news—I've added the implementation just like you asked.

However, I've made the decision to disable the discography feature by default. Why did I do that? Well, it turns out that the Deezer API we're using to fetch the artist's albums isn't the most reliable. To prevent any potential issues, I thought it would be safer to have the feature turned off initially.

But here's the frustrating part: the Deezer API doesn't provide access to the artist's full albums. I know, it's a huge letdown. As a result, the discography feature is a bit hit or miss. Users might not see all the albums they're expecting, and there could be some missing ones.

Thanks for your understanding, and I'm looking forward to hearing your thoughts

**Comment by lindely** [comment]
Hi,

May I assume that when you say the API _doesn't provide access to the artist's full albums_, one can assume they only show albums available on Deezer? Or does their API basically say _"I can give you their latest album, but I won't promise more than that"_? Otherwise I don't really understand, sorry.

Regardless, even if the feature is a partial one, if it can be enabled with some disclaimer note in the settings warning of a potential incomplete list, it'd be a nice addition for some of us; at least for me it would be :-)

Thanks again.

**Comment by n000tavailable** [comment]
The API does not include all available albums, even those that are fully accessible on Deezer. It seems to only provide the Top Albums selection

**Comment by n000tavailable** [comment]
I just did a few things (https://github.com/n000tavailable/tunedetective/commit/021f38df08670c9886130b6839aebb427dfcfc47). 

maybe you and everyone else reading wants to test this apk.
https://matrix.to/#/!BZkLjMgRpdhPnWWKvO:matrix.org/$DhdkbcpRBrrTYKDQYpN_-hDXFYI5816zMcmRXhBKSbo?via=matrix.org


This commit refactors the showArtistDiscography function to display the full discography of an artist, including albums, EPs, and singles. Previously, only the top albums were shown. The modifications include:

Updated the getArtistDiscography function to retrieve albums, EPs, and singles using separate API requests.
Implemented logic to eliminate duplicate entries in the discography.
Preferred albums over EPs and singles with the same name to avoid duplication.
Sorted the discography by release date in descending order.
Displayed the complete discography in an AlertDialog with a ListView.

**Comment by lindely** [comment]
I just tried it out and these are my initial thoughts:

- I know the results combine albums, EP's and singles. Would be nice if we could choose which types to see.
- Date or year of release would be really useful to see. Basically, the regular search result (cover + release date + track list button), but for all items in the list.
- When I search an artist f.e. from the search history and then want full discography, I have to type the artist name again.
- Some items say _Album details not found_ after clicking.

If the Deezer API allows for

**Comment by n000tavailable** [comment]
Will be a bit of work, but I'll try to take care of it ;)

**Comment by n000tavailable** [comment]
It is already taking shape
![Screenshot_20230612-220616](https://github.com/n000tavailable/tunedetective/assets/135447708/3ee0ac77-dedf-45e7-a3b3-f630a0aab519)

**Comment by abmaheswara** [comment]
It's great if users can access music through search results

**Comment by n000tavailable** [comment]
> ### Idea: favorite artists
> It would be very cool (and I hope not too difficult?) too have the possibility to set an artist as a favorite. When you open the app, it fetches the last album/single for those artists and shows them in a list in the home.
> 
> If there is a limit in the call to APIs, it could be set also for the favorites so that it is respected by the app. For example if Deezer only allows to retrieve 5 artists' info at a time, than people can only set 5 favorite artists (but it's still very useful in my opinion)
> 
> ### Linked idea: notification for new publications
> If the first one is possible, than I think it should be possible to also periodically check for new releases and send a notification.
> 
> By the way, thanks for your app, it's already very useful :heart: 
> 


I really really like this idea. It would be such a cool feature. I'll see what I can do. If the first feature is implemented one day, the second one will be easy

**Comment by n000tavailable** [comment]
![Screenshot_20230613-231320.png](https://github.com/n000tavailable/tunedetective/assets/135447708/002bd743-c67e-4ae6-bc0a-51fd3409688f)

![Screenshot_20230613-231329.png](https://github.com/n000tavailable/tunedetective/assets/135447708/32cb31c4-1a52-4192-b334-32b034dbfcef)

I have built in the feature. However, it leads to a small bug that I am trying to fix. Notifications do not exist yet. Notifications still need a lot of testing time, because music releases don't drop every day.

**Comment by n000tavailable** [comment]
Latest apk can be tested here:

https://matrix.to/#/!BZkLjMgRpdhPnWWKvO:matrix.org/$m57S1e2RTxkLV_WKrp_JS-XlpOLR0FoUTTlA_CNRnQk?via=matrix.org

**Comment by microchipster** [comment]
Hi! Great app idea. Just dropping a brainstorm here since you're soliciting:

If Deezer API is problematic, some alternative sources that I have found useful for finding new music:
1. Musicbrainz https://musicbrainz.org is an open database, see https://github.com/beetbox/beets for ideas on interfacing with it.
2. Wikipedia artist pages, wikipedia API could be used to scan for new releases, has pretty good artist coverage for artists that release albums. It's semi-structured data depending on the page.
3. Bandcamp and soundcloud can be queried for free to some degree.
4. Crowd source it by making a dedicated message board (in or out the app) where people can post similar-to-x or if-you-like-x-then-checkout-y type recommendations, or scrape existing message boards (e.g. reddit, but maybe lemmy or mastodon or matrix with the upcoming API changes).

Many sites do "new music friday" or "new music releases of the week". Great for fresh stuff. Could scrape those:
- https://open.spotify.com/playlist/37i9dQZF1DX4JAvHpjipBk
- https://newmusicfriday.com/
-  https://www.npr.org/sections/allsongs/606254804/new-music-friday

Another way I like to find new music is to look for live shows near me and listen semi-randomly. Artists that are touring are more often than not putting out fresh music, or at least are in the zeitgeist. It's also really satisfying if you listen and like it and then can immediately go see them perform live. This requires sharing coarse location and an information source for events, but not in association with your identity especially if you use a VPN or proxy, so still private enough I think. https://www.jambase.com is quite scrape-able (not ideal vs API obviously), probably some event services offer tier free APIs. Could even just use the list of artists touring in the most popular cities to avoid sharing location.

I realize most of these ideas are scraping oriented and could greatly complicate the app, but I think I would want the app to have two main goals: privacy and coverage. Privacy-wise, you can access data that is not available via a free API, and coverage-wise, even the paid services are often severely lacking data about more obscure but amazing artists or artists who they don't have rights to catalogue. Food for thought.

Anyway, happy tune hunting!

**Comment by n000tavailable** [comment]
Hey, thanks for the kind words.

I think your ideas are really good. However, I'm currently happy with the Deezer API, as it's quite accurate in terms of information, unlike other (open source or more privacy-friendly) alternatives.
For example, your first source is missing some releases.

About your 4th point, we currently have a Matrix group, but it is quite small.

https://matrix.to/#/#tunedetective:matrix.org

Maybe I could add something that shows the new releases on a Friday, but that would have to be from Deezer. I do not know yet if that is the goal of my mission. ;)

I will see what can be done.



Translated with www.DeepL.com/Translator (free version)

**Comment by n000tavailable** [comment]
Hey, thanks for the kind words.

I think your ideas are really good. However, I'm currently happy with the Deezer API, as it's quite accurate in terms of information, unlike other (open source or more privacy-friendly) alternatives.
For example, your first source is missing some releases.

About your 4th point, we currently have a Matrix group, but it is quite small.

https://matrix.to/#/#tunedetective:matrix.org

Maybe I could add something that shows the new releases on a Friday, but that would have to be from Deezer. I do not know yet if that is the goal of my mission. ;)

I will see what can be done.



Translated with www.DeepL.com/Translator (free version)

**Comment by ViktorKahu** [comment]
Hey,

This app has great potential, however, it's a bit green at the moment.

I think the app needs a homescreen that displays cards of artists you've added.

For example, I don't want to refer to search or search history every time, I just want to pick from a list of artists that I've added. Search history may contain some once-off searches that I don't neccessarely want to scroll through to find what I'm looking for.

**Comment by n000tavailable** [comment]
> Hey,
> 
> This app has great potential, however, it's a bit green at the moment.
> 
> I think the app needs a homescreen that displays cards of artists you've added.
> 
> For example, I don't want to refer to search or search history every time, I just want to pick from a list of artists that I've added. Search history may contain some once-off searches that I don't neccessarely want to scroll through to find what I'm looking for.

Great idea! I was thinking about something similar newpipe does. If you don't know newpipe, it's a open sourcr youtube client.
There you can swipe between lates videos, subs and more

**Comment by mrheat** [comment]
Hi i found your App lately but i would share some ideas i Had in mind :

1. Beeing able to Login with Premium Account (should unlock API) or even create "Developer App" to use the token 

2. Load local Android Media lib, and map that against Deezer lib, (to find high quality Versions)

3. You Could maybe think about A Download function to bulk replace all Low Bitrate Versions of your local Songs with Deezer Versions (selectable quality flac1411kbit or 320kmp3 i think.)

For me the use Case would rise enorm. If you experienced the Sound difference from the compressed mp3 or m4a stuff at 128k or even lower, against high Bitrate or even flac, your ears will be bleeding and you will be able to recognize Just by Hearing what quality IT IS 


PS Theres a Project called "deemix" still running fine but maybe IT could be of use somehow 

As far as i now there IS to Date No Other Deezer mobile Client as the original and this Project 

Great Project , keep IT Up 

Cheerz

**Comment by n000tavailable** [comment]
> Hi i found your App lately but i would share some ideas i Had in mind :
> 
> 1. Beeing able to Login with Premium Account (should unlock API) or even create "Developer App" to use the token 
> 
> 2. Load local Android Media lib, and map that against Deezer lib, (to find high quality Versions)
> 
> 3. You Could maybe think about A Download function to bulk replace all Low Bitrate Versions of your local Songs with Deezer Versions (selectable quality flac1411kbit or 320kmp3 i think.)
> 
> For me the use Case would rise enorm. If you experienced the Sound difference from the compressed mp3 or m4a stuff at 128k or even lower, against high Bitrate or even flac, your ears will be bleeding and you will be able to recognize Just by Hearing what quality IT IS 
> 
> 
> PS Theres a Project called "deemix" still running fine but maybe IT could be of use somehow 
> 
> As far as i now there IS to Date No Other Deezer mobile Client as the original and this Project 
> 
> Great Project , keep IT Up 
> 
> Cheerz

Really great idea, but I think that's not part of the projects goal to implement that and would be way to much work for a little hobby project.

**Comment by SantiagoPaak** [comment]
Hello! First I want to say thank you for the app. I wanted something like this and it helps a lot as it is.

A nice improvement however would be a general band info section instead of just the latest release. Some info on the band, socials, or whatever you can get available. It would also make more accessible the discography button without searching the band again.

A more important matter could be an import/export button, to keep the searched or watched bands between reinstalls of the app or moving phones.

Also a lil player instead of the tap/double tap thing would be nice.

As for the APIs, it might be a good idea to have access to other APIs instead of just Deezer. A simple dropdown that allows access to info of other APIs could be good enough. Just to have some extra options.

I think all of these would be nice improvements. Also I wanted to know if it is possible to contribute and give pull requests and stuff. Again, great work and I'll keep an eye on the progress. Cheers!

**Comment by n000tavailable** [comment]
> Hello! First I want to say thank you for the app. I wanted something like this and it helps a lot as it is.
> 
> A nice improvement however would be a general band info section instead of just the latest release. Some wikipedia info on the band, socials, or whatever you can get available. It would also make more accessible the discography button without searching the band again.
> 
> A more important matter could be an import/export button, to keep the searched or watched bands between reinstalls of the app or moving phones.
> 
> Also a lil player instead of the tap/double tap thing would be nice.
> 
> As for the APIs, it might be a good idea to have access to other APIs instead of just Deezer. A simple dropdown that allows access to info of other APIs could be good enough. Just to have some extra options. But I understand it might be hard to implement a whole extra API.
> 
> I think all of these would be nice improvements. Also I wanted to know if it is possible to contribute and give pull requests and stuff. Again, great work and I'll keep an eye on the progress. Cheers!  

Great ideas. I'll see which ones make it into the app, but all sound very usefull. Pull requests are of course possible and I would be happy about that.
